### PR TITLE
Remove equipments infos from ElasticSearch when removing variants

### PR DESCRIPTION
**Pull request by niconoir**

### CODE REVIEW

**Comment by SlimaneAmar** [comment]
I think the deleting of a variant like creation should be done on the network side with the same variant listener

File: src/main/java/org/gridsuite/study/server/NetworkVariantsListener.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        // Nothing to do in this listener
+    }
+
+    @Override
+    public void onUpdate(Identifiable identifiable, String attribute, Object oldValue, Object newValue) {
+        // Nothing to do in this listener
+    }
+
+    @Override
+    public void onVariantCreated(String sourceVariantId, String targetVariantId) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/NetworkVariantsListener.java:52]: Really necessary ?
Creation is done on the modification server side

Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/study/server/NetworkVariantsListener.java:52]: Is all the code added for the variant creation necessary ?

File: src/main/java/org/gridsuite/study/server/elasticsearch/EquipmentInfosServiceImpl.java [hunk 2]
```diff
@@ -53,15 +58,59 @@ public TombstonedEquipmentInfos addTombstonedEquipmentInfos(TombstonedEquipmentI
     }
 
     @Override
-    public Iterable<EquipmentInfos> findAllEquipmentInfos(@NonNull UUID networkUuid) {
+    public void addAllEquipmentInfos(List<EquipmentInfos> equipmentsInfos) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/elasticsearch/EquipmentInfosServiceImpl.java:61]: Why ?

File: src/main/java/org/gridsuite/study/server/elasticsearch/EquipmentInfosServiceImpl.java [hunk 3]
```diff
@@ -53,15 +58,59 @@ public TombstonedEquipmentInfos addTombstonedEquipmentInfos(TombstonedEquipmentI
...
+        Lists.partition(equipmentsInfos, partitionSize)
+                .parallelStream()
+                .forEach(equipmentInfosRepository::saveAll);
+    }
+
+    @Override
+    public void addAllTombstonedEquipmentInfos(List<TombstonedEquipmentInfos> tombstonedEquipmentInfos) {
+        Lists.partition(tombstonedEquipmentInfos, partitionSize)
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/elasticsearch/EquipmentInfosServiceImpl.java:69]: Why we need this service ?

File: src/main/java/org/gridsuite/study/server/elasticsearch/EquipmentInfosServiceImpl.java [hunk 4]
```diff
@@ -53,15 +58,59 @@ public TombstonedEquipmentInfos addTombstonedEquipmentInfos(TombstonedEquipmentI
...
+    @Override
+    public void deleteVariants(@NonNull UUID networkUuid, List<String> variantIds) {
+        variantIds.forEach(variantId -> {
+            equipmentInfosRepository.deleteAllByNetworkUuidAndVariantId(networkUuid, variantId);
+            tombstonedEquipmentInfosRepository.deleteAllByNetworkUuidAndVariantId(networkUuid, variantId);
+        });
+    }
+
+    @Override
+    public void cloneVariantModifications(@NonNull UUID networkUuid, @NonNull String variantToCloneId, @NonNull String variantId) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/elasticsearch/EquipmentInfosServiceImpl.java:93]: Remove

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/study/server/elasticsearch/EquipmentInfosRepository.java [hunk 5]
```diff
@@ -20,7 +21,11 @@
...
 @Lazy
 public interface EquipmentInfosRepository extends ElasticsearchRepository<EquipmentInfos, String> {
-    Iterable<EquipmentInfos> findAllByNetworkUuid(@NonNull UUID networkUuid);
+    List<EquipmentInfos> findAllByNetworkUuid(@NonNull UUID networkUuid);
+
+    List<EquipmentInfos> findAllByNetworkUuidAndVariantId(@NonNull UUID networkUuid, @NonNull String variantId);
 
     void deleteAllByNetworkUuid(UUID networkUuid);
+
+    void deleteAllByNetworkUuidAndVariantId(UUID networkUuid, String variantId);
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/study/server/elasticsearch/EquipmentInfosRepository.java:30]: Add @NonNull ?

File: src/main/java/org/gridsuite/study/server/elasticsearch/EquipmentInfosServiceMock.java [hunk 6]
```diff
@@ -32,21 +32,41 @@ public TombstonedEquipmentInfos addTombstonedEquipmentInfos(TombstonedEquipmentI
         return tombstonedEquipmentInfos;
     }
 
+    @Override
+    public void addAllEquipmentInfos(List<EquipmentInfos> equipmentsInfos) {
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/study/server/elasticsearch/EquipmentInfosServiceMock.java:36]: default implementation on interface EquipmentInfosService ?

File: src/main/java/org/gridsuite/study/server/elasticsearch/EquipmentInfosServiceMock.java [hunk 7]
```diff
@@ -32,21 +32,41 @@ public TombstonedEquipmentInfos addTombstonedEquipmentInfos(TombstonedEquipmentI
...
     }
 
+    @Override
+    public void deleteVariants(UUID networkUuid, List<String> variantIds) {
+        // Nothing to delete
+    }
+
+    @Override
+    public void cloneVariantModifications(UUID networkUuid, String variantToCloneId, String variantId) {
+        // Nothing to clone
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/study/server/elasticsearch/EquipmentInfosServiceMock.java:67]: same remark

File: src/main/java/org/gridsuite/study/server/elasticsearch/EquipmentInfosServiceMock.java [hunk 8]
```diff
@@ -32,21 +32,41 @@ public TombstonedEquipmentInfos addTombstonedEquipmentInfos(TombstonedEquipmentI
...
+    }
+
     @Override
     public List<EquipmentInfos> searchEquipments(@NonNull final String query) {
         return Collections.emptyList();
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/study/server/elasticsearch/EquipmentInfosServiceMock.java:72]: same remark

File: src/main/java/org/gridsuite/study/server/elasticsearch/EquipmentInfosServiceMock.java [hunk 9]
```diff
@@ -32,21 +32,41 @@ public TombstonedEquipmentInfos addTombstonedEquipmentInfos(TombstonedEquipmentI
...
 
     @Override
-    public Iterable<TombstonedEquipmentInfos> findAllTombstonedEquipmentInfos(@NonNull UUID networkUuid) {
+    public List<TombstonedEquipmentInfos> findAllTombstonedEquipmentInfos(@NonNull UUID networkUuid) {
         return Collections.emptyList();
     }
 
+    @Override
+    public void deleteVariants(UUID networkUuid, List<String> variantIds) {
+        // Nothing to delete
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/study/server/elasticsearch/EquipmentInfosServiceMock.java:62]: same remark

File: src/main/java/org/gridsuite/study/server/elasticsearch/EquipmentInfosServiceMock.java [hunk 10]
```diff
@@ -32,21 +32,41 @@ public TombstonedEquipmentInfos addTombstonedEquipmentInfos(TombstonedEquipmentI
...
 
     @Override
-    public Iterable<EquipmentInfos> findAllEquipmentInfos(@NonNull UUID networkUuid) {
+    public List<EquipmentInfos> findAllEquipmentInfos(@NonNull UUID networkUuid) {
         return Collections.emptyList();
     }
 
     @Override
-    public Iterable<TombstonedEquipmentInfos> findAllTombstonedEquipmentInfos(@NonNull UUID networkUuid) {
+    public List<TombstonedEquipmentInfos> findAllTombstonedEquipmentInfos(@NonNull UUID networkUuid) {
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/study/server/elasticsearch/EquipmentInfosServiceMock.java:56]: same remark

File: src/main/java/org/gridsuite/study/server/elasticsearch/TombstonedEquipmentInfosRepository.java [hunk 11]
```diff
@@ -20,7 +21,11 @@
...
 @Lazy
 public interface TombstonedEquipmentInfosRepository extends ElasticsearchRepository<TombstonedEquipmentInfos, String> {
-    Iterable<TombstonedEquipmentInfos> findAllByNetworkUuid(@NonNull UUID networkUuid);
+    List<TombstonedEquipmentInfos> findAllByNetworkUuid(@NonNull UUID networkUuid);
+
+    List<TombstonedEquipmentInfos> findAllByNetworkUuidAndVariantId(@NonNull UUID networkUuid, @NonNull String variantId);
 
     void deleteAllByNetworkUuid(UUID networkUuid);
+
+    void deleteAllByNetworkUuidAndVariantId(UUID networkUuid, String variantId);
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/study/server/elasticsearch/TombstonedEquipmentInfosRepository.java:30]: Add @NonNull ?

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/study/server/elasticsearch/EquipmentInfosServiceMock.java (hunk 6 above)
Comment (niconoir) [reply src/main/java/org/gridsuite/study/server/elasticsearch/EquipmentInfosServiceMock.java:36]: This is not really a default implementation, it is a specific implementation for EquipmentInfosServiceMock (which in fact does nothing).

File: src/main/java/org/gridsuite/study/server/elasticsearch/EquipmentInfosRepository.java (hunk 5 above)
Comment (niconoir) [reply src/main/java/org/gridsuite/study/server/elasticsearch/EquipmentInfosRepository.java]: Done

File: src/main/java/org/gridsuite/study/server/elasticsearch/TombstonedEquipmentInfosRepository.java (hunk 11 above)
Comment (niconoir) [reply src/main/java/org/gridsuite/study/server/elasticsearch/TombstonedEquipmentInfosRepository.java]: Done

File: src/main/java/org/gridsuite/study/server/NetworkVariantsListener.java (hunk 1 above)
Comment (niconoir) [reply src/main/java/org/gridsuite/study/server/NetworkVariantsListener.java]: Method removed.

File: src/main/java/org/gridsuite/study/server/elasticsearch/EquipmentInfosServiceImpl.java (hunk 2 above)
Comment (niconoir) [reply src/main/java/org/gridsuite/study/server/elasticsearch/EquipmentInfosServiceImpl.java]: Removed

File: src/main/java/org/gridsuite/study/server/elasticsearch/EquipmentInfosServiceImpl.java (hunk 3 above)
Comment (niconoir) [reply src/main/java/org/gridsuite/study/server/elasticsearch/EquipmentInfosServiceImpl.java]: Removed

File: src/main/java/org/gridsuite/study/server/elasticsearch/EquipmentInfosServiceImpl.java (hunk 4 above)
Comment (niconoir) [reply src/main/java/org/gridsuite/study/server/elasticsearch/EquipmentInfosServiceImpl.java]: Removed

**Review by FranckLecuyer — APPROVED** [review]

**Pull request merged**